kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java

```diff
@@ -85,9 +81,20 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
     }
 
     protected void close() {
+        assert !isActive.get();
+        // Clear the request queue
+        log.info("close channel {} with {} pending responses", ctx.channel(), requestQueue.size());
+        while (true) {
+            final ResponseAndRequest responseAndRequest = requestQueue.poll();
+            if (responseAndRequest != null) {
+                // Trigger writeAndFlushResponseToClient immediately, but it will do nothing because isActive is false
+                responseAndRequest.getResponseFuture().complete(null);
```

Comment:
May cause direct memory leak.

Reply:
It's a serious problem. I'll rethink how to resolve it.